launchflow/aws/lambda_service.py

```diff
@@ -1,88 +1,264 @@
+import inspect
+import os
+import shutil
+import subprocess
+import tempfile
 from dataclasses import dataclass
-from typing import List, Optional
+from typing import IO, Any, Dict, List, Optional, Union
 
+import requests
 from typing_extensions import Callable
 
 import launchflow as lf
 from launchflow import exceptions
-from launchflow.aws.acm import ACMCertificate
-from launchflow.aws.lambda_function import LambdaFunction
-from launchflow.aws.service import (
-    AWSDockerService,
-    AWSDockerServiceOutputs,
-    AWSStaticService,
+from launchflow.aws.api_gateway import (
+    CORS,
+    APIGateway,
+    APIGatewayLambdaIntegration,
+    APIGatewayRoute,
 )
+from launchflow.aws.lambda_function import (
+    LambdaFunction,
+    LambdaFunctionURL,
+    LambdaRuntime,
+)
+from launchflow.aws.service import AWSService
+from launchflow.config import config
 from launchflow.models.enums import ServiceProduct
+from launchflow.models.flow_state import AWSEnvironmentConfig
+from launchflow.models.launchflow_uri import LaunchFlowURI
 from launchflow.node import Inputs
 from launchflow.resource import Resource
-from launchflow.service import StaticServiceOutputs
+from launchflow.service import ServiceOutputs
+from launchflow.workflows.utils import zip_source
+
+
+def _get_relative_handler_import_path(func: Callable) -> str:
+    # Get the absolute path of the directory containing the launchflow.yaml file
+    launchflow_yaml_abspath = os.path.dirname(
+        os.path.abspath(config.launchflow_yaml.config_path)
+    )
+    # Get the absolute path of the file where the function is defined
+    func_file_path = os.path.abspath(inspect.getfile(func))
+    # Ensure the cwd has a trailing slash to avoid issues with relative path calculation
+    cwd = os.path.abspath(launchflow_yaml_abspath) + os.path.sep
+    # Convert the absolute path to a relative path with respect to the current working directory
+    relative_path = os.path.relpath(func_file_path, cwd)
+    # Convert the relative file path to a Python module path
+    module_path = relative_path.replace(os.path.sep, ".").rsplit(".py", 1)[0]
+    # Append the function name to the module path
+    full_import_path = f"{module_path}.{func.__name__}"
+
+    return full_import_path
+
+
+def _clean_pycache(directory: str):
```

Comment:
could this just be part of the build ignore?

Reply:
good thinking, updated